io/zenoh-transport/src/unicast/link.rs

```diff
@@ -82,6 +92,10 @@ impl TransportLinkUnicast {
                 queue_size: self.transport.config.manager.config.queue_size,
                 backoff: self.transport.config.manager.config.queue_backoff,
             };
+
+            #[cfg(feature = "transport_compression")]
+            let compression_enabled = self.transport.config.manager.config.compression_enabled;
```

Comment:
This could be removed and the configuration parameter passed directly at line `113`.

Reply:
The problem when doing that is that the function from the executor would take ownership of "self" (because of the async move), causing trouble later when self is required.